### PR TITLE
Rename `asm.dwarf` to `asm.debuginfo`

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

just link https://github.com/rizinorg/rizin/issues/907

### CODE REVIEW

File: librz/core/cconfig.c [hunk 1]
```diff
@@ -3029,10 +3029,10 @@ RZ_API int rz_core_config_init(RzCore *core) {
 	SETBPREF("asm.imm.trim", "false", "Remove all offsets and constants from disassembly");
 	SETBPREF("asm.indent", "false", "Indent disassembly based on reflines depth");
 	SETI("asm.indentspace", 2, "How many spaces to indent the code");
-	SETBPREF("asm.dwarf", "false", "Show dwarf comment at disassembly");
-	SETBPREF("asm.dwarf.abspath", "false", "Show absolute path in asm.dwarf");
-	SETBPREF("asm.dwarf.file", "true", "Show filename of asm.dwarf in pd");
-	SETBPREF("asm.dwarf.lines", "true", "Show DWARF source line information at disassembly");
+	SETBPREF("asm.debuginfo", "false", "Show debug info comment at disassembly");
+	SETBPREF("asm.debuginfo.abspath", "false", "Show absolute path in asm.debuginfo");
+	SETBPREF("asm.debuginfo.file", "true", "Show filename of asm.debuginfo in pd");
```
Comment (XVilka) [inline librz/core/cconfig.c]: Maybe `Show file name from the debug information in disassembly`?

File: librz/core/cconfig.c [hunk 2]
```diff
@@ -3029,10 +3029,10 @@ RZ_API int rz_core_config_init(RzCore *core) {
 	SETBPREF("asm.imm.trim", "false", "Remove all offsets and constants from disassembly");
 	SETBPREF("asm.indent", "false", "Indent disassembly based on reflines depth");
 	SETI("asm.indentspace", 2, "How many spaces to indent the code");
-	SETBPREF("asm.dwarf", "false", "Show dwarf comment at disassembly");
-	SETBPREF("asm.dwarf.abspath", "false", "Show absolute path in asm.dwarf");
-	SETBPREF("asm.dwarf.file", "true", "Show filename of asm.dwarf in pd");
-	SETBPREF("asm.dwarf.lines", "true", "Show DWARF source line information at disassembly");
+	SETBPREF("asm.debuginfo", "false", "Show debug info comment at disassembly");
+	SETBPREF("asm.debuginfo.abspath", "false", "Show absolute path in asm.debuginfo");
```
Comment (XVilka) [inline librz/core/cconfig.c]: `Show absolute path for file from the debug information`?

File: librz/core/cconfig.c [hunk 3]
```diff
@@ -3029,10 +3029,10 @@ RZ_API int rz_core_config_init(RzCore *core) {
 	SETBPREF("asm.imm.trim", "false", "Remove all offsets and constants from disassembly");
 	SETBPREF("asm.indent", "false", "Indent disassembly based on reflines depth");
 	SETI("asm.indentspace", 2, "How many spaces to indent the code");
-	SETBPREF("asm.dwarf", "false", "Show dwarf comment at disassembly");
-	SETBPREF("asm.dwarf.abspath", "false", "Show absolute path in asm.dwarf");
-	SETBPREF("asm.dwarf.file", "true", "Show filename of asm.dwarf in pd");
-	SETBPREF("asm.dwarf.lines", "true", "Show DWARF source line information at disassembly");
+	SETBPREF("asm.debuginfo", "false", "Show debug info comment at disassembly");
```
Comment (XVilka) [inline librz/core/cconfig.c]: `Show debug information in disassembly`

File: librz/core/cconfig.c [hunk 4]
```diff
@@ -3029,10 +3029,10 @@ RZ_API int rz_core_config_init(RzCore *core) {
...
+	SETBPREF("asm.debuginfo.abspath", "false", "Show absolute path in asm.debuginfo");
+	SETBPREF("asm.debuginfo.file", "true", "Show filename of asm.debuginfo in pd");
+	SETBPREF("asm.debuginfo.lines", "true", "Show debug info source line information at disassembly");
```
Comment (XVilka) [inline librz/core/cconfig.c]: `Shoe source line information from debug metadata at dissassembly`?

File: librz/core/disasm.c [hunk 5]
```diff
@@ -290,9 +290,9 @@ typedef struct {
 	// caches
 	char *_tabsbuf;
 	int _tabsoff;
-	bool dwarfFile;
-	bool dwarfAbspath;
-	bool dwarfShowLines;
+	bool asm_debuginfo_file;
```
Comment (XVilka) [inline librz/core/disasm.c]: maybe remove `asm_` prefix? It's a structure of the disasm state after all.

File: librz/core/disasm.c [hunk 6]
```diff
@@ -117,7 +117,7 @@ typedef struct {
 	int tracespace;
 	int cyclespace;
 	int show_indent;
-	bool show_dwarf;
+	bool asm_debuginfo;
```
Comment (XVilka) [inline librz/core/disasm.c]: Same

File: librz/core/disasm.c (hunk 5 above)
Comment (XVilka) [reply librz/core/disasm.c]: Also maybe make it a substructure instead? So that it could be modified/extended easily when necessary?

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by b1llow** [comment]
Or maybe we can use `RzBitVector` to store all the flags?

**Comment by XVilka** [comment]
@imbillow, why not just structure with bools? This structure should not have too many copies in the memory anyway - it's only used for disassembly.

**Comment by wargio** [comment]
i'm actually very ok with a bitmask

**Comment by b1llow** [comment]
> @imbillow, why not just structure with bools? This structure should not have too many copies in the memory anyway - it's only used for disassembly.

I'm just a bit obsessive-compulsive. When I think about how a bool in C takes up 1 byte, yet its information content is only 1 bit.

**Review by XVilka — APPROVED** [review]

**Pull request merged**